### PR TITLE
Closes #277

### LINKED ISSUE #277: Rules methods cleanup: NextBestInfTheory
Please edit appropriately below if needed.

To do:

- [x] Revise documentation
  - [x] RStudio outline section dividers (use `# function or class name ----`)
  - [x] Add lifecycle badge (as a default take "stable" for existing functions or classes)
  - [x] Replace `##'` by `#'`
  - [x] Add to `pkgdown.yml`
  - [x] Remove file history and the timestamp in comments on top of file
- [x] Add unit tests
  - [x] For classes:
    - [x] Test default constructor
    - [x] For reference classes: test methods
  - [x] For functions: as usual
- [x] Refactor as needed
  - [x] Code styling (indentation etc.) - install package `styler` and use corresponding add-in
  - [x] Keep externally used names so that we don't break downstream code
  - [x] Remove `keywords` tags
  - [x] For functions or methods:
    - [x] Add assertions for arguments using `checkmate` functions
    - [x] If it is too long, break up into pieces (rule of thumb: maximum 50 lines of code)
    - [x] Consider renaming internal variables etc. if needed
    - [x] Avoid explicit use of `return`
  - [x] For classes:
    - [x] move function definitions outside if possible (e.g. move validation function as `validity_subjects()` to `Data-validity.R` and have corresponding test file `test-Data-validity.R`)
    - [x] In `setClass` function, just use the `slots` instead of outdated `representation`
    - [x] Remove things like `validObject(.Data())`

### PR DESCRIPTION
**Pull request by wwojciech**
Closes #277

### CODE REVIEW

File: tests/testthat/test-helpers_rules.R [hunk 1]
```diff
@@ -1,11 +1,38 @@
...
+    h_info_theory_dist(0.2, c(0.4, 0.5), 1.2),
+    "Assertion on 'target' failed: Must have length 1."
+  )
+  expect_error(
+    h_info_theory_dist(0.2, 0.4, c(1.2, 1.6)),
+    "Assertion on 'asymmetry' failed: Must have length 1."
+  )
+})
+
+test_that("h_info_theory_dist trhows the error for wrong asymmetry", {
```
Comment (danielinteractive) [inline tests/testthat/test-helpers_rules.R:33]:
```suggestion
test_that("h_info_theory_dist throws the error for wrong asymmetry", {
```

File: R/helpers_rules.R [hunk 2]
```diff
@@ -6,14 +6,19 @@
...
-#' @param target (`numeric `)\cr target probability of a DLT.
+#' @param prob (`numeric`)\cr vector or matrix with probabilities of a DLT occurring.
+#' @param target (`number `)\cr single target probability of a DLT.
 #' @param asymmetry (`number`)\cr describes the rate of penalization
 #'   for overly toxic does, range 0 to 2.
 #'
 #' @export
 #' @examples
-#' h_info_theory_dist(c(0.5, 0.2), c(0.5, 0.1), 1.2)
+#' h_info_theory_dist(c(0.5, 0.2), 0.4, 1.2)
```
Comment (danielinteractive) [inline R/helpers_rules.R:16]: Makes sense, I guess the example worked before too because of vector calculations

**Review by danielinteractive — APPROVED** [review]
nice work, thanks @wwojciech !

File: R/helpers_rules.R (hunk 2 above)
Comment (wwojciech) [reply R/helpers_rules.R:16]: Yap, but I think at the moment we need to to be more secure rather than some extra flexibility, never used :)

**Pull request merged**